packages/components/src/components/switch/switch.stories.ts

```diff
@@ -0,0 +1,174 @@
+      },
+      constants: { type: 'attribute', name: 'disabled', value: true },
+      args
+    });
+  }
+};
+
+/**
+ * Use the `required` attribute to mark the element as required. This can be used for form validation purposes.
+ */
+
+export const Required = {
+  parameters: { controls: { exclude: ['required'] } },
+  render: (args: any) => {
+    return generateTemplate({
+      axis: {
+        y: { type: 'attribute', name: 'required' }
+      },
+      args
+    });
+  }
+};
+
+export const Checked = {
+  parameters: { controls: { exclude: ['checked'] } },
+  render: (args: any) => {
+    return generateTemplate({
+      axis: {
+        x: {
+          type: 'attribute',
+          name: 'disabled',
+          values: [false, true]
+        }
+      },
+      constants: { type: 'attribute', name: 'checked', value: true },
+      args
+    });
+  }
+};
+
+/**
+ * Test invalid state inside a form.
+ */
+
+export const Invalid = {
+  parameters: { controls: { exclude: ['required'] } },
+  render: (args: any) => {
+    return html`
+      <form>
+        ${generateTemplate({
+          args,
+          constants: [{ type: 'attribute', name: 'required', value: true }]
+        })}
+        <sd-button style="margin-top: 16px" type="submit">Submit</sd-button>
+      </form>
+    `;
+  },
+  play: async ({ canvasElement }: { canvasElement: HTMLUnknownElement }) => {
+    const el = canvasElement.querySelector('sd-button');
+    await waitUntil(() => el?.shadowRoot?.querySelector('button'));
+    await userEvent.type(el!.shadowRoot!.querySelector('button')!, '{return}', { pointerEventsCheck: 0 });
+  }
+};
+
+/**
+ * Use the `base`, `control--unchecked`, `control--checked`, `checked` and `label` part selectors to customize the checkbox.
+ */
+export const Parts = {
+  parameters: {
+    controls: {
+      exclude: [
+        'base',
+        'control',
+        'control--unchecked',
+        'control--checked',
+        'checked-icon',
+        'label',
+        'title',
+        'name',
+        'value',
+        'size',
+        'disabled',
+        'checked',
+        'form',
+        'required',
+        'default'
+      ]
+    }
+  },
+  render: (args: any) => {
+    return generateTemplate({
+      axis: {
+        y: {
+          type: 'template',
+          name: 'sd-switch::part(...){outline: solid 2px red}',
+          values: ['base', 'control', 'control--unchecked', 'control--checked', 'thumb', 'label'].map(part => {
+            return {
+              title: part,
+              value: `
+                <style>
+                    #part-${part} sd-switch::part(${part}){outline: solid 2px red};
+                    .hidden {display: none}
+                </style>
+                <div id="part-${part}">${checkboxTemplate(part)}</div>
+                <div class="hidden">%TEMPLATE%</div>
+              `
+            };
+          })
+        }
+      },
+      args
+    });
+  }
+};
+
+const checkboxTemplate = (part: string) => {
```

Comment:
It might be stupid, but no matter how easy something looks, we shouldn't forget to describe it 🙂